pkg/providers/cloudstack/validator_registry.go

```diff
@@ -0,0 +1,41 @@
+package cloudstack
+
+import (
+	"fmt"
+
+	"github.com/aws/eks-anywhere/pkg/executables"
+	"github.com/aws/eks-anywhere/pkg/filewriter"
+	"github.com/aws/eks-anywhere/pkg/networkutils"
+	"github.com/aws/eks-anywhere/pkg/providers/cloudstack/decoder"
+)
+
+// ValidatorRegistry exposes a single method for retrieving the CloudStack validator, and abstracts away how they are injected.
+type ValidatorRegistry interface {
+	Get(execConfig *decoder.CloudStackExecConfig) (*Validator, error)
+}
+
+// ValidatorFactory implements the ValidatorRegistry interface and holds the necessary structs for building fresh Validator objects.
+type ValidatorFactory struct {
+	builder     *executables.ExecutablesBuilder
```

Comment:
can we make this an interface so we don't depend on the executables package?

Reply:
I'm not sure I understand your suggestion. How do we remove the dependency on the executables package?

Reply:
ahh yeah
```go

type ClientBuilder interface {
    BuildCloudtsackClient(writer filewriter.FileWriter, skipIPCheck bool) Client
}

type ValidatorFactory struct {
	builder     ClientBuilder
	writer      filewriter.FileWriter
	skipIPCheck bool
}
```

Reply:
I was not able to get rid of the executables dependency. When I tried using the ProviderCmk interface (where you refer to `Client`), the ClientBuilder did not implement the interface properly

```
Cannot use 'builder' (type *executables.ExecutablesBuilder) 
as the type ClientBuilder Type does not implement 'ClientBuilder' 
need the method: BuildCmkExecutable(writer filewriter.FileWriter, config *decoder.CloudStackExecConfig) (ProviderCmkClient, error) 
have the method: BuildCmkExecutable(writer filewriter.FileWriter, config *decoder.CloudStackExecConfig) (*Cmk, error)
```

Using the original `Cmk` interface retains the dependency on the executables package, and plumbing down the ProviderCmk interface into the executables package doesn't make sense either.

Reply:
> plumbing down the ProviderCmk interface into the executables package doesn't make sense either

Ideally we shouldn't need to do that but IMO it's the lesser of two evils. I prefer to import an interface defined in the cloudstack package from the executables builder than using an executables concrete implementation from the cloudstack package. This follows inwards pointing dependency rule (your details [client implementation] can depend on your business logic but not the other way around).

If we want to avoid this it would require deeply refactoring the executables package (which I think it's too big), but that's another topic :)

Btw I would create a new method, keep `BuildCmkExecutable` as is and have a `BuildCloustackClient` that returns the interface.